src/app/components/Application/Main.jsx

```diff
@@ -1,33 +1,45 @@
+/* globals document */
 import React from 'react';
-import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 
 import BookList from '../BookList/BookList';
 import Sidebar from '../Sidebar/Sidebar';
 import utils from '../../utils/utils';
+import staffPicksDate from '../../utils/DateService';
+import appConfig from '../../../../appConfig';
 
 class Main extends React.Component {
   constructor(props) {
     super(props);
 
+    const picksData = this.props.picksData;
+    const picks =
+      this.filterByAudience(picksData.picks, this.props.currentAudience, picksData.type);
+
     this.state = {
       selectableFilters: this.props.selectableFilters,
       selectedFilters: [],
-      picks: this.props.currentPicks.picks && this.props.currentPicks.picks.length ?
-        this.props.currentPicks.picks : [],
+      picks,
     };
 
     this.setSelectedFilter = this.setSelectedFilter.bind(this);
     this.clearFilters = this.clearFilters.bind(this);
+    this.getPicksInfo = this.getPicksInfo.bind(this);
+    this.getNewPickSet = this.getNewPickSet.bind(this);
+    this.filterByAudience = this.filterByAudience.bind(this);
+    this.getCount = this.getCount.bind(this);
   }
 
-  componentWillReceiveProps() {
+  componentWillReceiveProps(nextProps) {
```

Comment:
@ktp242 I also noticed you set this up which is okay but it's not used properly, it should have some kind of logic check that returns true or false. If the `nextProps` don't match the current `props`, then you _want_ to update the component. Otherwise, if they're the same you do not need to update and re-render the component. We can update this later to conform to the lifecycle rule.